src/egon/data/processing/openstreetmap.py

```diff
@@ -0,0 +1,62 @@
+from egon.data import db
+import egon.data.config
+
+
+def modify_tables():
+    """Adjust primary keys, indices and schema of OSM tables.
+
+    * The Column "gid" is added and used as the new primary key.
+    * Indices (GIST, GIN) are reset
+    * The tables are moved to the schema configured as the "output_schema".
+    """
+    # Replace indices and primary keys
+    for table in [
+        "osm_" + suffix for suffix in ["line", "point", "polygon", "roads"]
+    ]:
+
+        # Drop indices
+        sql_statements = [f"DROP INDEX {table}_index;"]
+
+        # Drop primary keys
+        sql_statements.append(f"DROP INDEX {table}_pkey;")
+
+        # Add primary key on newly created column "gid"
+        sql_statements.append(f"ALTER TABLE public.{table} ADD gid SERIAL;")
+        sql_statements.append(
+            f"ALTER TABLE public.{table} ADD PRIMARY KEY (gid);"
+        )
+        sql_statements.append(
+            f"ALTER TABLE public.{table} RENAME COLUMN way TO geom;"
+        )
+
+        # Add indices (GIST and GIN)
+        sql_statements.append(
+            f"CREATE INDEX {table}_geom_idx ON public.{table} "
+            f"USING gist (geom);"
+        )
+        sql_statements.append(
+            f"CREATE INDEX {table}_tags_idx ON public.{table} "
+            f"USING GIN (tags);"
+        )
```

Comment:
Theres a name for the concept behind those changes. Updating the table schema like this is called a "migration". Theres a nice migration framework called "Alembic", built by the guy behind SQLAlchemy. It would probably be nice to figure out how to use that to encapsulate these changes. That's a TODO for myself though.

Reply:
I would love to learn more about migrations. Theoretically, I know what it does, but I've never used it. For now, I leave it on your TODO list ;-)

Reply:
Submitted as #55. See also the conversation above.